### PR TITLE
Add support for team repositories endpoint

### LINKED ISSUE #202: Add `list-team-repos`
Listing all the repos of a team is currently not possible. I am already working on a PR.

Corresponding GitHub API doc: https://developer.github.com/v3/orgs/teams/#list-team-repos

### PR DESCRIPTION
**Pull request by nblumoe**
closes #202

### CODE REVIEW

**Comment by nblumoe** [comment]
The failing build seems to be network hiccups. Not sure about the failing TLS installation though. Any hints?

File: samples/github-samples.cabal [hunk 1]
```diff
@@ -1,4 +1,4 @@
--- This file has been generated from package.yaml by hpack version 0.13.0.
+-- This file has been generated from package.yaml by hpack version 0.11.2.
```
Comment (phadej) [inline samples/github-samples.cabal]: Ah, it seems you use old `stack`. Update it, IIRC `stack-1.1.0` has `hpack-0.13.0` bundled in

File: src/GitHub/Endpoints/Organizations/Teams.hs [hunk 2]
```diff
@@ -139,6 +142,24 @@ listTeamMembersR tid r = PagedQuery ["teams", toPathPart tid, "members"] [("role
         TeamMemberRoleMaintainer  -> "maintainer"
         TeamMemberRoleMember      -> "member"
 
+-- | The repositories of a single team, by team id.
+-- | With authentication
+--
+-- > listTeamRepos' (Just $ OAuth "token") 1010101
```
Comment (phadej) [inline src/GitHub/Endpoints/Organizations/Teams.hs]: Does this work? IIRC one cannot create an id from the literal number?

File: src/GitHub/Endpoints/Organizations/Teams.hs [hunk 3]
```diff
@@ -139,6 +142,24 @@ listTeamMembersR tid r = PagedQuery ["teams", toPathPart tid, "members"] [("role
         TeamMemberRoleMaintainer  -> "maintainer"
         TeamMemberRoleMember      -> "member"
 
+-- | The repositories of a single team, by team id.
+-- | With authentication
```
Comment (phadej) [inline src/GitHub/Endpoints/Organizations/Teams.hs]: the bar on the second line is harmful.

File: src/GitHub/Endpoints/Organizations/Teams.hs [hunk 4]
```diff
@@ -139,6 +142,24 @@ listTeamMembersR tid r = PagedQuery ["teams", toPathPart tid, "members"] [("role
...
+--
+-- > listTeamRepos' (Just $ OAuth "token") 1010101
+listTeamRepos' :: Maybe Auth -> Id Team -> IO (Either Error (Vector Repo))
+listTeamRepos' auth tid = executeRequestMaybe auth $ listTeamReposR tid Nothing
+
+-- | Query team repositories
+-- See <https://developer.github.com/v3/orgs/teams/#list-team-repos>
+listTeamReposR :: Id Team -> Maybe Count -> Request k (Vector Repo)
+listTeamReposR team  = PagedQuery ["teams", toPathPart team, "repos"] []
```
Comment (phadej) [inline src/GitHub/Endpoints/Organizations/Teams.hs]: would use `tid` variable name here also

File: src/GitHub/Endpoints/Organizations/Teams.hs [hunk 5]
```diff
@@ -139,6 +142,24 @@ listTeamMembersR tid r = PagedQuery ["teams", toPathPart tid, "members"] [("role
...
         TeamMemberRoleMember      -> "member"
 
+-- | The repositories of a single team, by team id.
+-- | With authentication
+--
+-- > listTeamRepos' (Just $ OAuth "token") 1010101
+listTeamRepos' :: Maybe Auth -> Id Team -> IO (Either Error (Vector Repo))
+listTeamRepos' auth tid = executeRequestMaybe auth $ listTeamReposR tid Nothing
+
+-- | Query team repositories
```
Comment (phadej) [inline src/GitHub/Endpoints/Organizations/Teams.hs]: A ful-stop after the title. (Next line is rendered next to it, check `stack haddock`)

**Comment by nblumoe** [comment]
I "accidentally" fixed a few more docstrings in this commit now. Does not do any harm, however I decided not to fix them all, just so we don't have all those unrelated changes in this commit & PR.

**Comment by phadej** [comment]
Thanks!

**Pull request merged**